File: 455. Assign Cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());     
        int k = 0; 
        for (auto &x : s) {
            if (k < g.size() && g[k] <= x) ++k; 
        }
        return k; 
    }
};

File: 455. Assign Cookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int count = 0;
        int k = 0;
        for(int l = 0; k<g.length && l<s.length; l++){
            if(g [k] <= s[l]){
                k++;
            }
        }
        return k;
    }
}
